lib/mix/lib/mix/tasks/test.ex

```diff
@@ -560,12 +560,15 @@ defmodule Mix.Tasks.Test do
       :noop ->
         cond do
           opts[:stale] ->
+            ExUnit.Runner.run_no_tests()
             Mix.shell().info("No stale tests")
 
           files == [] ->
+            ExUnit.Runner.run_no_tests()
```

Comment:
What happens if we change this line:

https://github.com/jared-mackey/elixir/blob/ca8be02a9376fff3b4571a2dbd0ec1064c7fe103/lib/mix/lib/mix/compilers/test.ex#L41

To return `{:noop, ExUnit.run()}` and handle it here? Would it be ok, except that I guess it would print the summary?

Reply:
Actually, it could be:

    _ = ExUnit.run()
    :noop

Because if that works, I would prefer to still call ExUnit.run() but disable the formatter so nothing is logged. :)

Reply:
I’d think that would work fine. I’m not sure if anything in ExUnit is expecting there to be tests to run (since it hasn’t been running without any). But if there’s no assumptions then yeah, this would definitely be cleaner. It would be a bit more overhead since it starts and stops apps but should be negligible. 

I’m struggling testing this since my app doesn’t compile on 1.14 😢 (Tesla dep doesn’t compile)

Reply:
> I’m struggling testing this since my app doesn’t compile on 1.14 😢 (Tesla dep doesn’t compile)

Which error you get? that would be a regression we should address.

Reply:
@josevalim https://github.com/teamon/tesla/issues/524

Reply:
@jared-mackey can you please open up an issue here about it too?

Reply:
You got it.